consensus/diffs.go

```diff
@@ -27,133 +28,85 @@ type FileContractDiff struct {
-		if DEBUG {
-			_, exists := s.siacoinOutputs[scod.ID]
-			if exists {
-				panic("rogue new output in applyOutputDiff")
-			}
+// commitSiacoinOutputDiff applies or reverts a SiacoinOutputDiff.
+func (s *State) commitSiacoinOutputDiff(scod SiacoinOutputDiff, dir DiffDirection) {
+	// Sanity check - should not be adding an output twice, or deleting an
+	// output that does not exist.
+	if DEBUG {
+		_, exists := s.siacoinOutputs[scod.ID]
+		if exists == (scod.Direction == dir) {
+			panic("rogue output in commitSiacoinOutputDiff")
 		}
+	}
 
+	if scod.Direction == dir {
 		s.siacoinOutputs[scod.ID] = scod.SiacoinOutput
 	} else {
-		// Sanity check - output should exist.
-		if DEBUG {
-			_, exists := s.siacoinOutputs[scod.ID]
-			if !exists {
-				panic("rogue non-new output in applyOutputDiff")
-			}
-		}
-
 		delete(s.siacoinOutputs, scod.ID)
 	}
 }
 
-// commitFileContractDiff takes a FileContractDiff and applies it to the
-// consensus set.  `applied` indicates whether the diff is being applied or
-// reversed.
-func (s *State) commitFileContractDiff(fcd FileContractDiff, applied bool) {
-	add := fcd.New
-	if !applied {
-		add = !add
-	}
-
-	if add {
-		// Sanity check - contract should not already exist.
-		if DEBUG {
-			_, exists := s.fileContracts[fcd.ID]
-			if exists {
-				panic("rogue new contract in applyContractDiff")
-			}
+// commitFileContractDiff applies or reverts a FileContractDiff.
+func (s *State) commitFileContractDiff(fcd FileContractDiff, dir DiffDirection) {
+	// Sanity check - should not be adding a contract twice, or deleting a
+	// contract that does not exist.
+	if DEBUG {
+		_, exists := s.fileContracts[fcd.ID]
+		if exists == (fcd.Direction == dir) {
+			panic("rogue contract in commitFileContractDiff")
 		}
+	}
 
+	if fcd.Direction == dir {
 		s.fileContracts[fcd.ID] = fcd.FileContract
 	} else {
-		// Sanity check - contract should exist.
-		if DEBUG {
-			_, exists := s.fileContracts[fcd.ID]
-			if !exists {
-				panic("rogue non-new contract in applyContractDiff")
-			}
-		}
-
 		delete(s.fileContracts, fcd.ID)
 	}
 }
 
-// commitSiafundOutputDiff takes a SiafundOutputDiff and applies it to the
-// consensus set. `applied` indicates whether the diff is being applied or
-// reversed.
-func (s *State) commitSiafundOutputDiff(sfod SiafundOutputDiff, applied bool) {
-	add := sfod.New
-	if !applied {
-		add = !add
-	}
-
-	if add {
-		// Sanity check - output should not already exist.
-		if DEBUG {
-			_, exists := s.siafundOutputs[sfod.ID]
-			if exists {
-				panic("rogue new output in applyOutputDiff")
-			}
+// commitSiafundOutputDiff applies or reverts a SiafundOutputDiff.
+func (s *State) commitSiafundOutputDiff(sfod SiafundOutputDiff, dir DiffDirection) {
+	// Sanity check - should not be adding an output twice, or deleting an
+	// output that does not exist.
+	if DEBUG {
+		_, exists := s.siafundOutputs[sfod.ID]
+		if exists == (sfod.Direction == dir) {
+			panic("rogue output in commitSiafundOutputDiff")
```

Comment:
siafund output. This one I also try and be religious because 'output' all by itself typically refers to a siacoin output. Not too terrible to see 'output' all by itself if it's referring to a siacoin output but worse if it's referring to a siafund output.